src/python/pants/backend/project_info/tasks/filedeps.py

```diff
@@ -48,16 +53,20 @@ def console_output(self, targets):
     # Filter out any synthetic targets, which will not have a build_file attr.
     concrete_targets = {target for target in concrete_targets if not target.is_synthetic}
     for target in concrete_targets:
-      files.add(self._full_path(target.address.rel_path))
+      files.add(self._files_path(target.address.rel_path))
       if output_globs or target.has_sources():
         if output_globs:
           globs_obj = target.globs_relative_to_buildroot()
           if globs_obj:
-            files.update(self._full_path(src) for src in globs_obj['globs'])
+            files.update(self._files_path(src) for src in globs_obj['globs'])
         else:
-          files.update(self._full_path(src) for src in target.sources_relative_to_buildroot())
+          files.update(self._files_path(src) for src in target.sources_relative_to_buildroot())
       # TODO(John Sirois): BundlePayload should expose its sources in a way uniform to
       # SourcesPayload to allow this special-casing to go away.
       if isinstance(target, JvmApp) and not output_globs:
-        files.update(itertools.chain(*[list(bundle.filemap.keys()) for bundle in target.bundles]))
+        abs_path = self.get_options().abs
+        if abs_path:
+          files.update(itertools.chain(*[list(bundle.filemap.keys()) for bundle in target.bundles]))
```

Comment:
Rather than writing this whole line twice, let's maybe:
```python
filemap = bundle.filemap if self.get_options().absolute else bundle.relative_filemap
files.update(itertools.chain(*[list(filemap.keys()) for bundle in target.bundles]))
```